modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/ClayTable.tsx

```diff
@@ -0,0 +1,661 @@
+import {Body, Cell, Head, Row, Table} from '@clayui/core';
+import {ClayCheckbox, ClayRadio} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {FDSTableCellHTMLElementBuilderArgs} from '@liferay/js-api/data-set';
+import classNames from 'classnames';
+import {ClientExtension} from 'frontend-js-components-web';
+import {throttle} from 'frontend-js-web';
+import React, {useContext, useEffect, useMemo, useRef, useState} from 'react';
+
+import {IItemsActions} from '../..';
+import FrontendDataSetContext, {
+	IFrontendDataSetContext,
+	TRenderer,
+} from '../../FrontendDataSetContext';
+import Actions from '../../actions/Actions';
+import {getInternalCellRenderer} from '../../cell_renderers/getInternalCellRenderer';
+import {
+	ILocalizedItemDetails,
+	getLocalizedValue,
+} from '../../utils/getLocalizedValue';
+import {getInputRendererById} from '../../utils/renderer';
+import ViewsContext, {
+	IViewsContext,
+	TViewsContextDispatch,
+} from '../ViewsContext';
+
+// @ts-ignore
+
+import {VIEWS_ACTION_TYPES} from '../viewsReducer';
+import {InlineEditInputRenderer} from './TableCell';
+import TableContext from './dnd_table/TableContext';
+
+type Field = {
+	contentRenderer: string;
+	contentRendererClientExtension: boolean;
+	fieldName: any;
+	label: string;
+	localizeLabel: boolean;
+	mapData: Function;
+	sortable: boolean;
+};
+
+type Props = {
+	fields: Array<Field>;
+	inlineAddingSettings?: {
+		apiURL?: string;
+		defaultBodyContent?: Record<string, any>;
+	};
+	itemInlineChanges?: Array<any>;
+	items: Array<any>;
+	itemsActions: Array<IItemsActions>;
+	nestedItemsReferenceKey?: string;
+	selectItems: Function;
+	selectable?: boolean;
+	selectedItemsKey: string;
+	selectedItemsValue: any;
+	selectionType?: string;
+};
+
+type Sorting = {
+	column: React.Key;
+	direction: 'ascending' | 'descending';
+};
+
+const defaultAddItem = {
+	editable: true,
+	fieldName: 'add',
+	id: 'add',
+};
+
+const defaultAlwaysVisibleColumns = new Set(['select']);
+
+export function ClayTable({
+	fields,
+	inlineAddingSettings,
+	itemInlineChanges,
+	items,
+	itemsActions,
+	nestedItemsReferenceKey,
+	selectItems,
+	selectable,
+	selectedItemsKey,
+	selectedItemsValue,
+	selectionType,
+}: Props) {
+	const {itemsChanges, updateItem} = useContext(FrontendDataSetContext);
+	const [sort, setSort] = useState<Sorting | null>(null);
+
+	const filteredItems = useMemo(() => {
+		if (!sort) {
+			return items;
+		}
+
+		return items.sort((a, b) => {
+			const column = (sort.column as string).includes(',')
+				? (sort.column as string).split(',')
+				: (sort.column as string);
+			const aValue: ILocalizedItemDetails | null = getLocalizedValue(
+				a,
+				column
+			);
+			const bValue: ILocalizedItemDetails | null = getLocalizedValue(
+				b,
+				column
+			);
+
+			let cmp = new Intl.Collator('en', {numeric: true}).compare(
```

Comment:
Better to not abbreviate on the variable name.

Reply:
Also should 'en' be a locale variable that's passed in?

Reply:
> Also should 'en' be a locale variable that's passed in?

Hm yes, it's good when the data is written in another language, I'll check if this is really necessary and do some tests.